content/posts/pandastutor.md

```diff
@@ -0,0 +1,323 @@
+  podcast](https://talkpython.fm/episodes/show/358/understanding-pandas-visually-with-pandastutor)
+  for this suggestion!)
+- Caching file downloads: *[work-in-progress]* Some pandas code uses
+  functions like
+  [read_csv](https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html)
+  or
+  [read_html](https://pandas.pydata.org/docs/reference/api/pandas.read_html.html)
+  to download data files from the web. It would be great to cache those
+  files locally in-browser so that when the user re-runs that code in
+  the future, Pandas Tutor doesn't need to repeat the same
+  (potentially-large) downloads.
+- Start-up speed: *[work-in-progress]* Even if all packages are cached,
+  it still takes a few seconds to initialize Pandas Tutor at page load
+  time. We plan to profile our code and might try using
+  [pyc-wheel](https://pypi.org/project/pyc-wheel/) to compile `*.py` in
+  our wheel into `*.pyc` files.
+
+
+## The future: Pandas Tutor + Pyodide everywhere!
+
+Pyodide is really exciting for us since it opens the door to **embedding
+Pandas Tutor into any existing website** so that anyone can learn data
+science within the context of their favorite websites without installing
+anything.
+
+To achieve this, we're building a
+[bookmarklet](https://en.wikipedia.org/wiki/Bookmarklet) that injects
+Pandas Tutor + Pyodide into **any webpage** to add an inline code editor
+that can access its DOM. This will allow students to write Python code
+to, say, parse HTML tables using
+[BeautifulSoup](https://www.crummy.com/software/BeautifulSoup/), clean
+up that raw data, and plot it. Here's a mock-up screenshot of injecting
+Pandas Tutor (in the dashed red box) into a Wikipedia page to explore
+and plot country populations for a social studies class:
+
+<img src="pandastutor-wiki.png"/>
+
+(This idea originally came from the 2017 research paper, *DS.js: Turn
+Any Webpage into an Example-Centric Live Programming Environment for
+Learning Data Science* by Xiong Zhang and Philip Guo: [download
+PDF](https://pg.ucsd.edu/publications/DSjs-turn-any-webpage-into-data-science-IDE_UIST-2017.pdf))
+
+Next, we can take this idea even further ... wouldn't it be cool if the
+official [pandas API
+docs](https://pandas.pydata.org/docs/reference/index.html) were
+runnable? With this bookmarklet, we can inject Pandas Tutor + Pyodide
+into any API docs to automatically parse its code examples, visualize
+them, and let students edit that code live to explore different parameter
+values:
+
+<img src="pandastutor-api-docs.jpg"/>
+
+We can also do the same with Stack Overflow or other help forum sites
+because they often have self-contained code examples that can be
+directly visualized:
+
+<img src="pandastutor-stackoverflow.jpg"/>
+
+This is just the beginning of our journey with Pyodide. Stay tuned for
+more in the coming months as we use it to build tools to teach data
+science at scale!
+
+-- [Sam](https://www.samlau.me/) and [Philip](https://pg.ucsd.edu/)
```

Comment:
You can also add a longer bio as in other posts if you want, but this is also fine. And it's true that you already introduced yourself in the beginning.